src/main/java/org/graylog/integrations/ipfix/codecs/IpfixCodec.java

```diff
@@ -0,0 +1,212 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.integrations.ipfix.codecs;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
+import com.google.inject.assistedinject.Assisted;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.netty.buffer.Unpooled;
+import org.graylog.integrations.ipfix.Flow;
+import org.graylog.integrations.ipfix.InformationElementDefinitions;
+import org.graylog.integrations.ipfix.IpfixException;
+import org.graylog.integrations.ipfix.IpfixJournal;
+import org.graylog.integrations.ipfix.IpfixParser;
+import org.graylog.integrations.ipfix.TemplateRecord;
+import org.graylog2.plugin.Message;
+import org.graylog2.plugin.ResolvableInetSocketAddress;
+import org.graylog2.plugin.configuration.Configuration;
+import org.graylog2.plugin.configuration.ConfigurationRequest;
+import org.graylog2.plugin.configuration.fields.ConfigurationField;
+import org.graylog2.plugin.configuration.fields.TextField;
+import org.graylog2.plugin.inputs.annotations.Codec;
+import org.graylog2.plugin.inputs.annotations.ConfigClass;
+import org.graylog2.plugin.inputs.annotations.FactoryClass;
+import org.graylog2.plugin.inputs.codecs.AbstractCodec;
+import org.graylog2.plugin.inputs.codecs.CodecAggregator;
+import org.graylog2.plugin.inputs.codecs.MultiMessageCodec;
+import org.graylog2.plugin.inputs.transports.NettyTransport;
+import org.graylog2.plugin.journal.RawMessage;
+import org.joda.time.DateTime;
+import org.jooq.lambda.Seq;
+import org.jooq.lambda.tuple.Tuple2;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.net.InetSocketAddress;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Codec(name = "ipfix", displayName = "IPFIX Codec")
+public class IpfixCodec extends AbstractCodec implements MultiMessageCodec {
+    private static final Logger LOG = LoggerFactory.getLogger(IpfixCodec.class);
+
+    private final IpfixAggregator ipfixAggregator;
+    private final IpfixParser parser;
+    private InformationElementDefinitions infoElementDefs;
+
+    @Inject
+    protected IpfixCodec(@Assisted Configuration configuration, IpfixAggregator ipfixAggregator) {
+        super(configuration);
+        this.ipfixAggregator = ipfixAggregator;
+        infoElementDefs = new InformationElementDefinitions(
+                Resources.getResource("ipfix-iana-elements.json"),
```

Comment:
Can a comment be added here to clarify/indicate what these files are for? This might help to remind us what these are in the future. :)

I understand that these might become configurable later. If so, no need for a details comment if the literal strings are going away anyhow. :)

Reply:
great idea! comments have been added. :)